File: hw2/codes/gon_furkan.rb
for i in 0..4 do
  puts("------------------------------")
  # shows unlabeled exit
  # break statement exits only one loop
  for j in 0..4 do
      break if j==1
      puts("innerLoopBreakUnlabeled: ", j)
  end
  

puts("------------------------------")

  # no labeled exit in ruby
 
  # shows unlabeled continue 
  # continue statement affects only one loop
  for j in 0..4 do
      next if j==2
      puts("innerLoopContinueUnlabeled: ", j)
  end
  puts("------------------------------")

  # no labeled continue in ruby

  # no labeled exit in ruby
end

File: hw2/codes/gon_furkan.php
<?php

    for ($i = 0; $i < 5; $i++) { 
     
     echo "-------------------------------\n";
   
     
     // shows unlabeled exit
     // break statement exits only one loop
    for ($j = 0; $j < 5; $j++) {
       if ($j==1) break;
       echo("innerLoopBreakUnlabeled: $j \n");
    }// end of for
    
     
    echo("-------------------------------\n");
     
     
     // shows labeled exit
     // break statement exits only one loop
    for ($j = 0; $j < 5; $j++){
       if ($j==1) break 1;
       echo("innerLoopBreakLabeled: $j \n");
    }// end of for
     
    echo("-------------------------------\n");
    
     
     // shows unlabeled continue 
     // continue statement affects only one loop
    for ($j = 0; $j < 5; $j++){
       if($j==2) continue;
       print("innerLoopContinueUnlabeled: $j \n");
    }// end of for
     
    echo("-------------------------------\n");
    
    // shows labeled continue
    // continue statement affects only one loop
    for ($j = 0; $j < 5; $j++){
       if($j==2) continue 1;
       print("innerLoopContinueLabeled: $j \n");
    }// end of for
     
    echo("-------------------------------\n");
    
     
    // shows labeled exit for more than one loop
    // break statement exits from outerLoop and ends the program
    for ($j = 0; $j < 5; $j++){
       if($j==3) break 2;
       echo("outerLoopBreakLabeled: $j \n");
    }// end of for 
     
   }// end of for
?>
